### ISSUE TITLE
Manage nuget packages solution level : update packages should not trigger selection "packages.config" or "packagereference" dialog

### ISSUE THREAD

**Issue by Styxxy** [NuGet/Home#7252]
## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe): **VS UI**
NuGet version (x.x.x.xxx): the one provided with VS
VS version (if appropriate): **15.7.5**
OS version (i.e. win10 v1607 (14393.321)): **Windows 7**

Worked before? If so, with which NuGet version: **no**

## Detailed repro steps so we can see the same problem

1. Having a solution with two projects. Both are non-SDK style projects.

2. One project has a nuget package installed but not the latest version (example: Newtonsoft.Json v10.0.3). The other project doesn't have any nuget package installed.

3. Manage nuget packages for the solution.

4. Go to the **Updates** tab.

5. Select the package to update, and press the Update button.

-->

6. A diolog pop-ups to select the installation mode for nuget packages for the project that doesn't have any packages installed.
**Expected**: don't show this dialog, we are just updating (and this process should respect the already chosen mode) and not newly installing any packages.

![nuget-update-dialog-format](https://user-images.githubusercontent.com/1635825/44849045-20f82400-ac59-11e8-9e43-843974c939e2.png)

### Sample Project

Sample repository: https://github.com/Styxxy/Issue-Nuget-DarkTheme

**Comment by StingyJack** [comment]
Would you expect this set of steps to not make any changes to teh ClassLibrary project? (no adding of app.config, reformatting files, changing the proj file, etc?)

**Comment by Styxxy** [comment]
@StingyJack Why would you need to make changes to a project that doesn't have: 1) any nuget packages, 2) is not being targeted, and 3) in a package **update** process (not install), thus only projects are being targeted that have already package(s) installed.

**Comment by StingyJack** [comment]
I'm asking because nuget has always done those things, which are mostly unexpected and unwanted to me as well.

**Comment by Styxxy** [comment]
@StingyJack
Yes indeed, for that I made this issue. It is quite unlogical and also adds much more (unneeded) dialogues to click through. Especially if you have to do this in many solution files :-) .

Sorry for my little bit too to the point answer earlier, I didn't have much time and wanted to add a response. I didn't mean to be as direct as it is written.

**Comment by PatoBeltran** [comment]
Hey @Styxxy, thanks for opening the issue. This does seem like a bad UX. I tried reproing this with the repository you provided and had no luck.

One easy workaround for this would be to make sure the checkbox for `Allow format selection on first package install` is not checked, this will avoid showing you that dialog if you don't plan to install packages in project B.

**Comment by Styxxy** [comment]
@PatoBeltran Thanks for the feedback.
Normally I would disable the "format selection", however I still need to support both types (and also within new development).

**Comment by StingyJack** [comment]
@PatoBeltran - what does an option for formatting have to do with package installation? and what selection would it be formatting?

**Comment by PatoBeltran** [comment]
@StingyJack The "allow format selection on first package install" means that in the first time a project interacts with nuget (i.e. installing a package), nuget will prompt to ask if that project should use PackageReferences or Packages.config. This is there to give the opportunity to the user to choose a package format on the user's first interaction with NuGet.

As I said in my previous comment, I agree that the behavior @Styxxy shows is not desirable because the prompt should not appear in solution level PM if you are only interacting with projects that already have packages installed.

**Comment by StingyJack** [comment]
So it means "Allow for format selection...". Got it.
